lapce-ui/src/editor/tab_header_content.rs

```diff
@@ -456,3 +517,114 @@ impl Widget<LapceTabData> for LapceEditorTabHeaderContent {
         }
     }
 }
+
+fn get_truncated_path(full_paths: &[PathBuf]) -> Vec<PathBuf> {
```

Comment:
Is this ```PathBuf.parent()```?

Reply:
Not quite, you can imagine it as consuming all the given `Path`s in parallel from left and right until it finds a mismatch `Component`. Just a simple algorithm to find a short unique path to display. I can't find the algorithm that VSCode it using so this is the best I can think of.